<!-- Launches delta and two image windows-->

<launch>
	<!-- User arguments -->

	<arg name="isBoardConnected" default="true" doc="Specify whether the arduino board is connected or not."/>
	<arg name="port" default="/dev/ttyACM1"/>
	<arg name="baud" default="57600"/>
	<arg name="video_device"  default="/dev/video1" />
	<arg name="show_image"  default="true" />
	<arg name="name" default="delta"/>

	<!-- camera capture -->
	<group if="$(arg isBoardConnected)">
		<!-- rosparam command="load" file="$(find lookat_firmware)/config/arduino_params.yaml" / -->
		<node ns="delta" name="arduino_node" pkg="rosserial_python" type="serial_node.py" output="screen">
			<param name="~port" value="$(arg port)" />
	        <param name="~baud" value="$(arg baud)" />
		</node>
	</group>


	<node ns="delta" required="true" name="delta_kinematics" pkg="delta" type="delta_kinematics" respawn="false" output="screen"/>

  <!--<node ns="delta" required="true" name="delta_control" pkg="delta" type="delta_control" respawn="false" output="screen"/><node ns="delta" required="true" name="delta_control" pkg="delta" type="delta_control" respawn="false" output="screen"/>-->
	<node ns="delta" required="true" name="delta_controlVector" pkg="delta" type="delta_controlVector" respawn="false" output="screen"/>


	<node
		name="usb_cam"
		pkg="usb_cam"
		type="usb_cam_node"
		output="screen" >
		<param name="video_device" value="$(arg video_device)" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap"/>
	</node>




	<!-- Provides rectified image, given raw image and camera info -->
	<node
		name="image_proc"
		pkg="image_proc"
		type="image_proc"
		ns="usb_cam"
		output="screen"
		respawn="true" >
	</node>



	<!--Fake processor-->
	<node name="delta"
		pkg="delta"
		type="delta"
		output="screen"
		args="1">
		<remap from="/delta/image_in" to="/usb_cam/image_rect_color"/>
		<remap from="/delta/camera_info_in" to="/usb_cam/camera_info"/>
	</node>

	<!-- display in/out image -->
	<group if="$(arg show_image)">
		<node
			name="image_view_raw"
			pkg="image_view"
			type="image_view"
			respawn="false"
			output="screen">
			<remap from="image" to="/usb_cam/image_raw"/>
			<param name="autosize" value="false" />
		</node>
		<node
			name="image_view_out"
			pkg="image_view"
			type="image_view"
			respawn="false"
			output="screen">
			<remap from="image" to="/delta/image_out"/>
			<param name="autosize" value="false" />
		</node>
	</group>

</launch>
